### ISSUE TITLE
from mayavi import mlab  generates ModuleNotFoundError: No module named 'vtkCommonCorePython'

### ISSUE THREAD

**Issue by BrazilForever11** [bfieldtools/bfieldtools#41]
I ran 

`from mayavi import mlab`

and got back: 

```
ImportError                               Traceback (most recent call last)
D:\user_data\Program_Files\Anaconda3\envs\tensorflow\lib\site-packages\vtkmodules\vtkCommonCore.py in <module>
      4     # use relative import for installed modules
----> 5     from .vtkCommonCorePython import *
      6 except ImportError:

ImportError: DLL load failed: The specified module could not be found.

During handling of the above exception, another exception occurred:

ModuleNotFoundError                       Traceback (most recent call last)
~\AppData\Local\Temp/ipykernel_3128/4106049245.py in <module>
----> 1 from mayavi import mlab

D:\user_data\Program_Files\Anaconda3\envs\tensorflow\lib\site-packages\mayavi\mlab.py in <module>
     14 # Mayavi imports
     15 from mayavi.core.common import process_ui_events
---> 16 from mayavi.tools.camera import view, roll, yaw, pitch, move
     17 from mayavi.tools.figure import figure, clf, gcf, savefig, \
     18     draw, sync_camera, close, screenshot

D:\user_data\Program_Files\Anaconda3\envs\tensorflow\lib\site-packages\mayavi\tools\camera.py in <module>
     22 # We can't use gcf, as it creates a circular import in camera management
     23 # routines.
---> 24 from .engine_manager import get_engine
     25 
     26 

D:\user_data\Program_Files\Anaconda3\envs\tensorflow\lib\site-packages\mayavi\tools\engine_manager.py in <module>
     11 # Local imports
     12 from mayavi.preferences.api import preference_manager
---> 13 from mayavi.core.registry import registry
     14 from mayavi.core.engine import Engine
     15 from mayavi.core.off_screen_engine import OffScreenEngine

D:\user_data\Program_Files\Anaconda3\envs\tensorflow\lib\site-packages\mayavi\core\registry.py in <module>
     14 
     15 # Local imports.
---> 16 from mayavi.core.metadata import Metadata, import_symbol
     17 
     18 # A logger for this module.

D:\user_data\Program_Files\Anaconda3\envs\tensorflow\lib\site-packages\mayavi\core\metadata.py in <module>
      9 # Enthought library imports.
     10 from traits.api import HasTraits, Str, Callable, Either, List, Instance
---> 11 from mayavi.core.pipeline_info import PipelineInfo
     12 
     13 

D:\user_data\Program_Files\Anaconda3\envs\tensorflow\lib\site-packages\mayavi\core\pipeline_info.py in <module>
     10 from traits.api import HasTraits, Enum, List
     11 
---> 12 from .utils import get_tvtk_dataset_name
     13 
     14 # The list of datasets supported.

D:\user_data\Program_Files\Anaconda3\envs\tensorflow\lib\site-packages\mayavi\core\utils.py in <module>
      1 import numpy as np
----> 2 import vtk
      3 from vtk.numpy_interface import dataset_adapter as dsa
      4 from vtk.numpy_interface import algorithms as algs
      5 

D:\user_data\Program_Files\Anaconda3\envs\tensorflow\lib\site-packages\vtk.py in <module>
     30     all_spec = importlib.util.find_spec('vtkmodules.all')
     31     all_m = importlib.util.module_from_spec(all_spec)
---> 32     all_spec.loader.exec_module(all_m)
     33 
     34     # import vtkmodules

D:\user_data\Program_Files\Anaconda3\envs\tensorflow\lib\site-packages\vtkmodules\all.py in <module>
      5 
      6 # --------------------------------------
----> 7 from .vtkCommonCore import *
      8 from .vtkCommonMath import *
      9 from .vtkCommonMisc import *

D:\user_data\Program_Files\Anaconda3\envs\tensorflow\lib\site-packages\vtkmodules\vtkCommonCore.py in <module>
      7     # during build and testing, the modules will be elsewhere,
      8     # e.g. in lib directory or Release/Debug config directories
----> 9     from vtkCommonCorePython import *

ModuleNotFoundError: No module named 'vtkCommonCorePython'
```

I have python 3.7.11, conda 4.10.3. vtk is installed in conda and I have folder with vtk and conda and conda is in the PATH variable

**Comment by anttimc** [comment]
Hi, I think the main problem is this:

ImportError: DLL load failed: The specified module could not be found.

Mayavi depends on Vtk which has some stuff complied for particular architechture, operating system etc. In this case python cannot find a dynamic library. I think you need to at least reinstall VTK. 

A better option could be that you create a new conda environment and install mayavi there first. After that install bfieldtools using pip in the conda env.

**Comment by anttimc** [comment]
Related #15

**Comment by jiivana** [comment]
Hello, When I was installing mayavi to a Windows machine I had similar problems and did not figure out a fix. 

I ended up installing MNE Python (https://mne.tools/stable/index.html) first with its conda environment (see https://mne.tools/stable/install/index.html). That environment has mayavi installed on it. After installation, I used that environment to install bfieldtools using pip and everything is working.
